tools/globcheck/globcheck.py

```diff
@@ -0,0 +1,147 @@
+    chopping = ChoppingPdbResLabel.from_str(f"{trimming_range}")
+
+    model_id = os.path.splitext(os.path.basename(pdb_file))[0]
+    domain_id = GeneralDomainID(raw_id=model_id, chopping=chopping)
+
+    parser = PDBParser(QUIET=True)
+    structure = parser.get_structure(model_id, pdb_file)
+
+    normed_rg = calculate_normed_radius_of_gyration(domain_id, structure, volume_resolution=5)
+    packing_density = calculate_packing_density(domain_id, structure, include_all_atoms=False, distance_cutoff=5)
+    log.info(f"✅ Normalized Rg = {normed_rg:.3f}, Packing Density = {packing_density:.3f}")
+
+    return normed_rg, packing_density
+
+def process_results(model_id, asphericity, normed_rg, packing_density, output_file):
+    """Processes results to finalise an output
+    file in CSV format. Adding column asserting
+    Pass/Fail of thresholds.
+
+    Args:
+        model_id (str): Identifier for the model
+        asphericity (flot): Asphericity value
+        normed_rg (float): Normalised rg value
+        packing_density (float): Packing Density
+        output_file (str): Path to output file 
+    """
+    result = {
+        "model_id": model_id,
+        "asphericity": asphericity,
+        "normalized_radius_gyration": normed_rg,
+        "packing_density": packing_density,}
+
+    df = pd.DataFrame([result])
+
+    df['satisfies_thresholds'] = (
+        (df['asphericity'] < asphericity_threshold) &
+        (df['normalized_radius_gyration'] < rg_threshold) &
+        (df['packing_density'] > pd_threshold))
+
+    df.to_csv(output_file, index=False, float_format="%.3f")
+    log.info(f"✅ Final results written to {output_file}")
+
+    row = df.iloc[0]
+    status = "✅ PASS" if row['satisfies_thresholds'] else "❌ FAIL"
+    log.info(f"{status}: {model_id} | Asp = {row['asphericity']:.3f}, Rg = {row['normalized_radius_gyration']:.3f}, PD = {row['packing_density']:.3f}")
+
+def main():
+    warnings.filterwarnings("ignore", category=UserWarning)
+    warnings.filterwarnings("ignore", category=DeprecationWarning)
+
+    parser = argparse.ArgumentParser(description="Run globularity metrics using CATH AlphaFlow and MDAnalysis.")
+    parser.add_argument("-p", "--pdb", required=True, help="Input trimmed PDB file")
+    parser.add_argument("-r", "--range", required=True, help="Residue trimming range (e.g., 160-214)")
+    parser.add_argument("-o", "--output", help="Output CSV file")
+    args = parser.parse_args()
+
+    model_id = os.path.splitext(os.path.basename(args.pdb))[0]
+
+    output_file = args.output or f"{model_id}_globularity_recap.csv"
+
+    try:
+        validate_residue_range(args.pdb, args.range)
+        asphericity = calculate_asphericity(args.pdb)
+        normed_rg, packing_density = calculate_globularity_metrics(args.pdb, args.range)
+        process_results(model_id, asphericity, normed_rg, packing_density, output_file)
+
+    except Exception as e:
+        log.error(f"An error occured {e}")
+        sys.exit(1)
```

Comment:
this style of exception handling is not very useful, it doesn't give any information either to the user or developer to understand what the problem was. Since we're not really handling exceptions in here I would rather just have the program crash, at least we get the traceback